maple2latex/src/translator.py

```diff
@@ -68,6 +72,126 @@ def __init__(self, inp):
             self.general = [self.fields["even"], self.fields["odd"]]
 
 
+class LatexEquation(object):
+    def __init__(self, label, equation, metadata):
+        self.label = label
+        self.equation = equation
+        self.metadata = metadata
+
+    @classmethod
+    def from_maple(cls, eq):
+        # modify fields
+        eq.lhs = translate(eq.lhs)
+        eq.factor = translate(eq.fields["factor"])
+        eq.front = translate(eq.fields["front"])
+
+        equation = eq.lhs + "\n  = "
+
+        if eq.factor == "1":
+            eq.factor = ""
+
+        metadata = dict()
+
+        # translates the Maple information (with spacing)
+        if eq.eq_type == "series":
+            eq.general = translate(eq.general[0])
+
+            if eq.factor != "":
+                equation += eq.factor + " "
+
+            elif eq.front != "":
+                equation += eq.front + "+"
+
+            equation += "\\sum_{k=0}^\\infty "
+
+            if eq.category == "power series":
+                equation += eq.general
+            elif eq.category == "asymptotic series":  # make sure to fix asymptotic series
+                equation += "(" + eq.general + ")"
+
+        elif eq.eq_type == "contfrac":
+            forms = list()
+
+            # substitution handling
+            if len(parse_brackets(eq.general[0])) > 1:
+                forms = parse_brackets(eq.general[0])
+
+            if len(eq.general) == 2 or forms:
+                if not forms:  # forms is empty
+                    for form in eq.general:
+                        forms += parse_brackets(form)
+
+                metadata["substitution"] = ','.join(perform_substitution(eq, forms))
+                pieces = ["s_m", "1"]
+
+            else:
+                pieces = parse_brackets(eq.general[0])[0]
+
+            if eq.begin != "":
+                eq.begin = parse_brackets(eq.begin)
+
+            start = 1  # in case the value of start isn't assigned
+
+            # add terms before general
+            if eq.front != "":
+                equation += eq.front + "+"
+                start = 1
+
+            if eq.begin != "":
+                for piece in eq.begin:
```

Comment:
Is `eq.begin` supposed to be a `str`? If that's the case, the directly preceding line is entirely unnecessary; the loop won't do anything if `eq.begin == ""`, because it has a length of `0`.